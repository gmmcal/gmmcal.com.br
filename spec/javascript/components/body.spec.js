import { mount } from '@vue/test-utils'
import Body from '@/components/body'
import About from '@/components/about'
import Contact from '@/components/contact'
import Educations from '@/components/educations'
import Experiences from '@/components/experiences'
import Skills from '@/components/skills'
import create from '../settings/utils'

describe('Body', () => {
  let wrapper

  beforeEach(() => {
    window.gon = {
      about: create('about'),
      educations: create('educations'),
      experiences: create('experiences'),
      skills: create('skills')
    }

    wrapper = mount(Body)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = []
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Child components', () => {
    it('includes About', () => {
      const child_component = wrapper.find(About)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('includes Contact', () => {
      const child_component = wrapper.find(Contact)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('includes Educations', () => {
      const child_component = wrapper.find(Educations)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('includes Experiences', () => {
      const child_component = wrapper.find(Experiences)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('includes Skills', () => {
      const child_component = wrapper.find(Skills)
      expect(child_component.isVueInstance()).toBeTruthy()
    })
  })
})
