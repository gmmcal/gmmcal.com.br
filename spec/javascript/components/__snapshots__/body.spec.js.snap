// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders correctly 1`] = `
<section>
  <section id="about" class="about-me">
    <div class="container">
      <div class="heading">
        <h2>About Me</h2>
      </div>
      <div class="row">
        <div class="col-xs-12 col-sm-12 col-md-7 col-lg-7 pb-5">
          <h3>Gustavo Cunha</h3>
          <h4 class="sub-heading">Some title at some company</h4>
          <div class="description">
            <p>Lorem ipsum nulus dolor.</p>
          </div> <a href="file.pdf" target="_blank" rel="noopener" class="btn-download">PDF Download</a>
        </div>
        <div class="col-xs-12 col-sm-12 col-md-5 col-lg-4 offset-lg-1">
          <div class="lazy topmar"></div>
        </div>
      </div>
    </div>
  </section>
  <section id="experience" class="experience">
    <div class="container">
      <div class="heading">
        <h2>Work Experience</h2>
        <p>Companies that I have worked for</p>
      </div>
      <div class="row work-details">
        <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
          <div class="year">Jan / 2015<br>Jan / 2016</div>
        </div>
        <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
          <div class="arrowpart"></div>
          <div class="ex-con">
            <h4><span>Amazing company</span> - <span><span>Great city</span> - <span>Some country</span></span></h4>
            <h5>Ninja developer</h5>
            <div class="description">
              <p>Lorem ipsum nulus dolor</p>
            </div>
          </div>
        </div>
      </div>
      <div class="row work-details">
        <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
          <div class="year">Jan / 2015<br>Jan / 2016</div>
        </div>
        <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
          <div class="arrowpart"></div>
          <div class="ex-con">
            <h4><span>Amazing company</span> - <span><span>Great city</span> - <span>Some country</span></span></h4>
            <h5>Ninja developer</h5>
            <div class="description">
              <p>Lorem ipsum nulus dolor</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section id="skills" class="technical">
    <div class="container">
      <div class="heading">
        <h2>Technical Skills</h2>
        <p>The subjects I'm currently under my radar, but I'm always open to new technologies.</p>
      </div>
      <div class="row">
        <p class="skill" style="font-size: 70px;">Skill Name</p>
        <p class="skill" style="font-size: 63px;">Another Skill</p>
      </div>
    </div>
  </section>
  <section id="education" class="education">
    <div class="container">
      <div class="heading">
        <h2>Education and Diplomas</h2>
        <p>What I did on my academic career</p>
      </div>
      <div class="row education-details">
        <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
          <div class="year">Currently</div>
        </div>
        <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
          <div class="arrowpart"></div>
          <div class="ex-con">
            <h4>Some awesome university</h4>
            <h5>Computer science</h5>
            <div class="description">
              <p>Lorem ipsum nulus dolor</p>
            </div>
          </div>
        </div>
      </div>
      <div class="row education-details">
        <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
          <div class="year">Jan / 2016</div>
        </div>
        <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
          <div class="arrowpart"></div>
          <div class="ex-con">
            <h4>Some awesome university</h4>
            <h5>Computer science</h5>
            <div class="description">
              <p>Lorem ipsum nulus dolor</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section id="contact">
    <section class="contact">
      <div class="container top-con">
        <div class="heading">
          <h2>Contact me</h2>
          <p>Feel free to contact me</p>
        </div>
      </div>
    </section>
    <section class="contact-details">
      <div class="container">
        <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12">
          <h4>Contacts</h4>
          <div class="row">
            <p class="location col-xs-12 col-sm-6 col-md-6 col-lg-4"><svg aria-hidden="true" data-prefix="fas" data-icon="map-marker-alt" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512" prefix="fas" class="svg-inline--fa fa-map-marker-alt fa-w-12">
                <path fill="currentColor" d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z" class=""></path>
              </svg> <span>City</span> - <span>Country</span></p>
            <p class="phone_number col-xs-12 col-sm-6 col-md-6 col-lg-4"><svg aria-hidden="true" data-prefix="fas" data-icon="mobile-alt" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512" prefix="fas" class="svg-inline--fa fa-mobile-alt fa-w-10">
                <path fill="currentColor" d="M272 0H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h224c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zM160 480c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm112-108c0 6.6-5.4 12-12 12H60c-6.6 0-12-5.4-12-12V60c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v312z" class=""></path>
              </svg>
              817634986124
            </p>
            <p class="email col-xs-12 col-sm-6 col-md-6 col-lg-4"><svg aria-hidden="true" data-prefix="far" data-icon="envelope" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" prefix="fas" class="svg-inline--fa fa-envelope fa-w-16">
                <path fill="currentColor" d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z" class=""></path>
              </svg>
              email@domain.com
            </p>
            <p class="github col-xs-12 col-sm-6 col-md-6 col-lg-4"><svg aria-hidden="true" data-prefix="fab" data-icon="github" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512" prefix="fab" class="svg-inline--fa fa-github fa-w-16">
                <path fill="currentColor" d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z" class=""></path>
              </svg> <a href="https://github.com/gmmcal" target="_blank" rel="noopener">gmmcal</a></p>
            <p class="linkedin col-xs-12 col-sm-6 col-md-6 col-lg-4"><svg aria-hidden="true" data-prefix="fab" data-icon="linkedin-in" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" prefix="fab" class="svg-inline--fa fa-linkedin-in fa-w-14">
                <path fill="currentColor" d="M100.3 480H7.4V180.9h92.9V480zM53.8 140.1C24.1 140.1 0 115.5 0 85.8 0 56.1 24.1 32 53.8 32c29.7 0 53.8 24.1 53.8 53.8 0 29.7-24.1 54.3-53.8 54.3zM448 480h-92.7V334.4c0-34.7-.7-79.2-48.3-79.2-48.3 0-55.7 37.7-55.7 76.7V480h-92.8V180.9h89.1v40.8h1.3c12.4-23.5 42.7-48.3 87.9-48.3 94 0 111.3 61.9 111.3 142.3V480z" class=""></path>
              </svg> <a href="https://linkedin.com/in/gmmcal" target="_blank" rel="noopener">Gustavo Cunha</a></p>
            <p class="twitter col-xs-12 col-sm-6 col-md-6 col-lg-4"><svg aria-hidden="true" data-prefix="fab" data-icon="twitter" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" prefix="fab" class="svg-inline--fa fa-twitter fa-w-16">
                <path fill="currentColor" d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z" class=""></path>
              </svg> <a href="https://twitter.com/gmmcal" target="_blank" rel="noopener">@gmmcal</a></p>
          </div>
        </div>
      </div>
    </section>
  </section>
</section>
`;
